File: todo-reservation-springboot/src/main/java/de/vkb/dojo/es/reservations/model/state/Reservation.java
package de.vkb.dojo.es.reservations.model.state;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import de.vkb.dojo.es.reservations.model.view.PersonData;
import de.vkb.dojo.es.reservations.model.view.RoomData;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Reservation {
    private final RoomData room;
    private final PersonData person;

    @JsonCreator
    public Reservation(
            @JsonProperty("room") RoomData room,
            @JsonProperty("person") PersonData person
    ) {
        this.room = room;
        this.person = person;
    }

    @JsonGetter
    public RoomData getRoom() {
        return room;
    }

    @JsonGetter
    public PersonData getPerson() {
        return person;
    }
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/reservations/model/state/Reservation.kt
package de.vkb.dojo.es.reservations.model.state

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import de.vkb.dojo.es.reservations.model.view.PersonData
import de.vkb.dojo.es.reservations.model.view.RoomData

@JsonIgnoreProperties(ignoreUnknown = true)
data class Reservation(
    val person: PersonData,
    val room: RoomData
)
